### DISCUSSION TITLE
Implement "goto type definition"

### DISCUSSION THREAD

**Discussion by tekumara** [microsoft/pylance-release#2618] [Enhancement]
Is it possible to go to an inferred type definition?

eg: 
In this example I'm hovering over the `encryption` variable and can see the type is `GetBucketEncryptionOutputTypeDef`.
I'd like to go to the definition of the type, so I can see it in all its glory.

<img width="946" alt="Screen Shot 2020-11-16 at 10 50 17 am" src="https://user-images.githubusercontent.com/125105/99200490-bca1ea80-27f9-11eb-939e-7e3145aadb38.png">

**Comment by jakebailey** [comment]
Originally, I thought this would need a custom call and UI, but LSP 3.16 (which we use already) supports go-to-type-def.

**Comment by debonte** [comment]
Support for Go To Type Definition was added in Pylance [2021.12.0](https://github.com/microsoft/pylance-release/releases/tag/2021.12.0). I'm locking this discussion. Comments can be added on this issue if needed: https://github.com/microsoft/pylance-release/issues/2859
